packages/@ngtools/webpack/src/loader.ts

```diff
@@ -274,9 +274,36 @@ function _getResourceRequest(element: ts.Expression, sourceFile: ts.SourceFile)
 function _replaceResources(refactor: TypeScriptFileRefactor): void {
   const sourceFile = refactor.sourceFile;
 
+  _getResourceNodes(refactor)
+    // Get the full text of the initializer.
+    .forEach((node: ts.PropertyAssignment) => {
+      const key = _getContentOfKeyLiteral(sourceFile, node.name);
+
+      if (key == 'templateUrl') {
+        refactor.replaceNode(node,
+          `template: require(${_getResourceRequest(node.initializer, sourceFile)})`);
+      } else if (key == 'styleUrls') {
+        const arr = <ts.ArrayLiteralExpression[]>(
+          refactor.findAstNodes(node, ts.SyntaxKind.ArrayLiteralExpression, false));
+        if (!arr || arr.length == 0 || arr[0].elements.length == 0) {
+          return;
```

Comment:
Should we change this to `return [];`? If you have a `styleUrls: []` in your Component definition, this will cause `ng` to fail compilation.